dll/win32/iphlpapi/iphlpapi_main.c

```diff
@@ -2620,12 +2622,93 @@ SetIpForwardEntryToStack(PMIB_IPFORWARDROW pRoute)
 }
 
 DWORD WINAPI
-NhGetInterfaceNameFromDeviceGuid(DWORD dwUnknown1,
-                                 DWORD dwUnknown2,
-                                 DWORD dwUnknown3,
+NhGetInterfaceNameFromDeviceGuid(GUID * pInterfaceGUID,
+                                 PWCHAR pInterfaceName,
+                                 PULONG pOutBufLen,
                                  DWORD dwUnknown4,
                                  DWORD dwUnknown5)
 {
-    FIXME("NhGetInterfaceNameFromDeviceGuid() stub\n");
-    return 0L;
+    LPOLESTR pStr;
+    WCHAR szKeyName[2*MAX_PATH];
+    HKEY hKey;
+    DWORD cbData = 0, type = REG_SZ;
+
+    if (StringFromCLSID(pInterfaceGUID, &pStr) != ERROR_SUCCESS)
+    {
+        // failed to convert guid to string
+        return ERROR_INVALID_PARAMETER;
+    }
+
+    StringCbPrintfW(szKeyName, sizeof(szKeyName), L"SYSTEM\\CurrentControlSet\\Control\\Network\\{4D36E972-E325-11CE-BFC1-08002BE10318}\\%s\\Connection", pStr);
```

Comment:
What is this guid?
Also, you do not handle the return

Reply:
`Network Adapter Class = Net ClassGuid = {4d36e972-e325-11ce-bfc1-08002be10318}` according to MSDN.

I ported the code from here: https://github.com/reactos/reactos/blob/c2c66aff7dacc62d125f2cd61d1167e9a2aa3fd6/dll/shellext/netshell/lanconnectui.cpp#L386